book/accessibility.md

```diff
@@ -2122,99 +2122,161 @@ also defining new and fully stylable [form control elements][openui].
 
 :::
 
+
 Mixed voice / visual interaction
 ================================
 
-The accessibility tree also needs access to the geometry of each object. This
-allows screen readers to know where things are on the screen in case
-the user wants to [hit test][hit-test] a place on the screen to see what is
-there:  user who can't see the screen still might want to do things like touch
-exploration of the screen, or being notified what is under the mouse as they
-move it around.
+Thanks to our work in this chapter, our rendering pipeline now
+basically have two different outputs: a display list for visual
+interaction, and an accessibility tree for screen-reader interaction.
+Many users will use just one or the other. However, it can also be
+valuable to use both together. For example, a user might have limited
+vision, able to make out the general items on a web page but unable to
+read the text. Such a user might use their mouse to navigate the page,
+but need the items under the mouse to be read to them by a
+screen-reader.
+
+Implementing this feature will require each accessibility node to know
+about its geometry on the page. The user could then instruct the
+screen-reader to determine which object is under the mouse (this is
+called [hit testing][hit-test]) and read it aloud.
+
+[hit-test]: https://chromium.googlesource.com/chromium/src/+/HEAD/docs/accessibility/browser/how_a11y_works_3.md#Hit-testing
 
+Getting access to the geometry is tricky, because the accessibility
+tree is generated from the element tree, while the geometry is
+accessible in the layout tree. Let's add a `layout_object` pointer to
+each `Element` object:[^if-has]
 
-To get access to the geometry, let's add a `layout_object` pointer to each
-`Node` object if it has one. That's easy to do in the constructor of each layout
-object type. Here are two examples (don't forget to handle all of them):
+[^if-has]: If it has a layout object, that is. Some `Element`s might
+    not, and their `layout_object` pointers will stay `None`.
 
 ``` {.python}
-class DocumentLayout:
-    def __init__(self, node):
+class Element:
+    def __init__(self, tag, attributes, parent):
         # ...
-        node.layout_object = self
+        self.layout_object = None
+
+class Text:
+    def __init__(self, text, parent):
+        # ...
+        self.layout_object = None
 ```
 
+Now, when we construct a layout object, we can fill in the
+`layout_object` field of its `Element`. In `BlockLayout`, it looks
+like this:
+
 ``` {.python}
+class Element:
```

Comment:
remove?

Reply:
Why? You mean, just don't show the code?

Reply:
My inclination is to show it. If it's a bit obvious, that also means it'll be easy to read, which means the cost of showing it is low, but the benefit to the uncareful reader is potentially large.

Reply:
This code block says:

class Element:
 
class BlockLayout:
   .. some code

Why is the "class Element" part there? not sure what it indicates.

Reply:
Oh wow, that was such a crazy mistake I totally missed it. Thanks!!!